SystemManager/Inc/SM_States.hpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * SM_States.hpp
+ *
+ * System Manager State-Machine header file
+ * 
+ * Created on: Nov 7, 2022
+ * Author(s): Gordon Fountain
+ *
+ * Describes the states of SM:
+ * 
+ * Boot (executes only once) -> Disarm -> Ground Ops -> Flight
+ * 
+ * Also holds fatalFailure to shut down all other systems.
+ *
+ */
+
+#ifndef ZPSW3_SM_STATES_HPP
+#define ZPSW3_SM_STATES_HPP
+
+#include "SM_StateManager.hpp"
+#include "AM_DataTypes.hpp"
+#include "cmsis_os.h"
+
+namespace SM {
+
+class BootMode : public SystemState {
+    public:
+        void enter(SystemManager *system_manager) {(void) system_manager;}
+        void execute(SystemManager *system_manager);
+        void exit(SystemManager *system_manager) {(void) system_manager;}
+        static SystemState& getInstance();
+    private:
+        BootMode() {
+        }
```

Comment:
I recently learned you can also use delete instead of making these private
```suggestion
        BootMode() = delete;
```

Reply:
Compiler disagrrees
![image](https://user-images.githubusercontent.com/78159714/206822678-eb6aeb69-6134-44de-af94-36d73495e073.png)

Reply:
Ah, you are using the default compiler under the hood, so you're right it would need to be defined privately. Alternatively you can use:
```suggestion
        BootMode() = default;
```

I'll let you decide if you prefer `BootMode() {};`